analyses/tp53_nf1_score/05-evaluate-classifier.py

```diff
@@ -65,45 +65,30 @@
 np.random.seed(123)
 
 # read TP53/NF1 alterations
-status_df = pd.read_table(status_file, low_memory=False)
+full_status_df = pd.read_table(status_file, low_memory=False)
 # read in clinical file
 clinical_df = pd.read_table(clinical)
 # select only IDs
 clinical_df = clinical_df[
     ["Kids_First_Biospecimen_ID", "sample_id", "Kids_First_Participant_ID"]
 ]
 
-
-# add clinical info to alterations dataframe
-status_df = status_df.merge(
-    clinical_df,
-    how="left",
-    left_on="Tumor_Sample_Barcode",
-    right_on="Kids_First_Biospecimen_ID",
-)
-
-# Value count of variant classification
-print(status_df.Variant_Classification.value_counts())
-
-
 # Obtain a binary status matrix
-full_status_df = pd.crosstab(
-    status_df["sample_id"], status_df.Hugo_Symbol, dropna=False
-)
-full_status_df.head(3)
-full_status_df[full_status_df > 1] = 1
-full_status_df = full_status_df.reset_index()
-full_status_df = full_status_df.drop(["No_TP53_NF1_alt"], axis=1)
+# Obtain a binary status matrix
+for idx, val in enumerate(full_status_df.itertuples()):
+    # if both SNV/CNV counts are 1 or more than 1 this sample can be considered as a TP53 double hit
+    # if SNV or CNV has 1 hit with Li-Fraumeni syndrome cancer predisposition can be considered as double hit
+    # if SNV has more than 1 hit sample can be considered as double hit
+    if full_status_df.loc[idx, 'SNV_indel_counts'] >= 1 and full_status_df.loc[idx,'CNV_loss_counts'] >= 1 or full_status_df.loc[idx,'SNV_indel_counts'] == 1 and (full_status_df.loc[idx,'cancer_predispositions'] == "Li-Fraumeni syndrome") or full_status_df.loc[idx,'CNV_loss_counts'] == 1 and (full_status_df.loc[idx,'cancer_predispositions'] == "Li-Fraumeni syndrome") or full_status_df.loc[idx,'SNV_indel_counts'] >1:
```

Comment:
Actually, can you split this onto multiple lines as well? I missed how very long this got...

Reply:
Thanks for the review, I've updated to multi lines and grouped with parenthesis.